mercantile/__init__.py

```diff
@@ -341,45 +338,140 @@ def parent(*tile):
+
+    x, y, z = tile
+    if x != int(x) or y != int(y) or z != int(z):
+        raise ValueError("Cannot find the parent of a fractional tile!")
+    target_zoom = z - 1 if zoom is None else zoom
+
+    # Algorithm heavily inspired by https://github.com/mapbox/tilebelt.
+    return_tile = tile
+    while return_tile[2] > target_zoom:
+        xtile, ytile, ztile = return_tile
+        if xtile % 2 == 0 and ytile % 2 == 0:
+            return_tile = Tile(xtile // 2, ytile // 2, ztile - 1)
+        elif xtile % 2 == 0:
+            return_tile = Tile(xtile // 2, (ytile - 1) // 2, ztile - 1)
+        elif not xtile % 2 == 0 and ytile % 2 == 0:
+            return_tile = Tile((xtile - 1) // 2, ytile // 2, ztile - 1)
+        else:
+            return_tile = Tile((xtile - 1) // 2, (ytile - 1) // 2, ztile - 1)
+    return return_tile
+
+
+def children(*tile, **kwargs):
+    """Get the children of a tile
 
     Parameters
     ----------
     tile : Tile or sequence of int
         May be be either an instance of Tile or 3 ints, X, Y, Z.
+    zoom : int, optional
+        Returns all children at zoom *zoom*, in depth-first clockwise winding order.
+        If unspecified, returns the immediate (i.e. zoom + 1) children of the tile.
 
     Returns
     -------
     list
     """
+    zoom = kwargs.get("zoom", None)
+
     if len(tile) == 1:
         tile = tile[0]
-    xtile, ytile, zoom = tile
-    return [
-        Tile(xtile * 2, ytile * 2, zoom + 1),
-        Tile(xtile * 2 + 1, ytile * 2, zoom + 1),
-        Tile(xtile * 2 + 1, ytile * 2 + 1, zoom + 1),
-        Tile(xtile * 2, ytile * 2 + 1, zoom + 1)]
+    if len(tile) == 2:
+        raise ValueError(
+            "Could not parse tile! Make sure that if you are calling this with zoom, you call this with zoom as a keyword argument."
+        )
+    xtile, ytile, ztile = tile
+
+    if zoom is not None and (ztile > zoom or zoom != int(zoom)):
+        raise ValueError(
+            "zoom must be an integer and greater than the source tile!")
+
+    target_zoom = zoom if zoom is not None else ztile + 1
+
+    queue = [tile]
+    while queue[0][2] < target_zoom:
+        xtile, ytile, ztile = queue.pop(0)
+        queue += [
+            Tile(xtile * 2, ytile * 2, ztile + 1),
+            Tile(xtile * 2 + 1, ytile * 2, ztile + 1),
+            Tile(xtile * 2 + 1, ytile * 2 + 1, ztile + 1),
+            Tile(xtile * 2, ytile * 2 + 1, ztile + 1)
+        ]
+    return queue
+
+
+def simplify(*tiles):
+    """Reduces the size of the tileset as much as possible by merging leaves into parents.
+
+    Parameters
+    ----------
+    tiles : Sequence of tiles to merge.
+
+    Returns
+    -------
+    list
+    """
+
+    def merge(merge_set):
+        """Checks to see if there are 4 tiles in merge_set which can be merged.
+        If there are, this merges them.
+        This returns a list of tiles, as well as a boolean indicating if any were merged.
```

Comment:
Is this inconsistent with the return value documented above?

Reply:
I don't think so - merge is a helper function that gets repeatedly applied to the tileset and isn't meant to be called by outside code.